### ISSUE TITLE
运行代码的rec.py 文件产生的checkpoint文件放到evaluation.py 进行评估测试，结果都是0

### ISSUE THREAD

**Issue by Quan-Yao** [rutgerswiselab/GenRec#2]
![03](https://github.com/rutgerswiselab/GenRec/assets/149983492/2ea4c186-f209-4558-a545-4c828a2a6b38)
![02](https://github.com/rutgerswiselab/GenRec/assets/149983492/49fa8b65-664d-41b1-964e-d5ffe662fe22)
![01](https://github.com/rutgerswiselab/GenRec/assets/149983492/df9897fe-a4be-4c3f-9366-edd73012627f)
运行的数据集是本代码中data里面的movies数据集
运行rec.py文件产生的output_dir = checkpoint_new，程序跑完产生文件checkpoint_new/checkpoint-200 将这个产生的文件放到evaluation.py中，将参数lora_weights 置为checkpoint_new/checkpoint-200 ，发现评估结果都是·0。这个是什么问题导致的？

**Comment by jianchaoji** [comment]
你训练完以后adapter 的大小是多少

**Comment by Quan-Yao** [comment]
> 你训练完成以后适配器的大小是多少

大小是1K，不知道是训练的时候哪里出现了问题。
![04](https://github.com/rutgerswiselab/GenRec/assets/149983492/e87e0c47-6e5a-4da9-b704-b1772e499388)

**Comment by jianchaoji** [comment]
可能是package版本问题，你把rec.py这几行注释掉再试试
`old_state_dict = model.state_dict `
    `model.state_dict = (`
        `lambda self, *_, **__: get_peft_model_state_dict(`
            `self, old_state_dict()`
       ` )`
    `).__get__(model, type(model))`

**Comment by Quan-Yao** [comment]
> 可能是包版本问题，你把rec.py这几行注释掉再尝试 `old_state_dict = model.state_dict ` `model.state_dict = (` `lambda self, *_, **__: get_peft_model_state_dict(` `self, old_state_dict()` ` )` `).__get__(model, type(model))`

注释这段代码以后，生产的适配器大小16.4MB，但把训练后的适配器放入评估代码中，评估指标还是0。 
![图片1](https://github.com/rutgerswiselab/GenRec/assets/149983492/25459aff-738b-41d2-9eec-e7199d7f703b)
![图片2](https://github.com/rutgerswiselab/GenRec/assets/149983492/87006e49-2709-478b-88f8-940450745c6f)
![图片3](https://github.com/rutgerswiselab/GenRec/assets/149983492/5bbd211f-9830-4c66-b91c-c2c1774ae05d)

**Comment by jianchaoji** [comment]
你能把你训练完的结果打印一两个吗

**Comment by Quan-Yao** [comment]
> 你能把训练完成的结果打印两个吗？

movies测试集中，第一个样本放入到评估里面
-------label-------（真实的输出）
Eternal Sunshine of the Spotless Mind (2004)
--------res---------（产生的output）
['Léon: The Professional (a.k.a. The Professional) (Léon) (a.k.a. The Professional) (Léon: The Professional) (Léon) (Léon: The Professional) (Léon) (Léon: The Professional) (Léon) (L', 'Léon: The Professional (a.k.a. The Professional) (Léon) (a.k.a. The Professional) (Léon: The Professional) (Léon) (Léon: The Professional) (Léon) (Léon) (Léon) (Léon) (', 'Léon: The Professional (a.k.a. The Professional) (Léon) (a.k.a. The Professional) (Léon: The Professional) (Léon) (Léon: The Professional) (Léon) (Léon: The Professional) (Léon: The Professional', 'Léon: The Professional (a.k.a. The Professional) (Léon) (a.k.a. The Professional) (Léon: The Professional) (Léon) (Léon: The Professional) (Léon: The Professional) (Léon: The Professional) (Léon', 'Léon: The Professional (a.k.a. The Professional) (Léon) (a.k.a. The Professional) (Léon: The Professional) (Léon) (Léon: The Professional) (Léon) (Léon) (Léon: The Professional) (L', 'Léon: The Professional (a.k.a. The Professional) (Léon) (a.k.a. The Professional) (Léon: The Professional) (Léon) (Léon) (Léon: The Professional) (Léon) (Léon: The Professional) (L', 'Léon: The Professional (a.k.a. The Professional) (Léon) (a.k.a. The Professional) (Léon: The Professional) (Léon) (Léon) (Léon: The Professional) (Léon) (Léon) (Léon) (', 'Léon: The Professional (a.k.a. The Professional) (Léon) (a.k.a. The Professional) (Léon: The Professional) (Léon) (Léon) (Léon: The Professional) (Léon: The Professional) (Léon) (L', 'Léon: The Professional (a.k.a. The Professional) (Léon) (a.k.a. The Professional) (Léon: The Professional) (a.k.a. The Professional) (Léon) (a.k.a. The Professional) (Léon)', 'Léon: The Professional (a.k.a. The Professional) (Léon) (a.k.a. The Professional) (Léon: The Professional) (Léon) (Léon: The Professional) (Léon) (Léon) (Léon) (Léon)']

在generate.py文件下，产生的output为：
Sixth Sense, The (1999)u
Sixth Sense, The (1999)
Sixth Sense, The (1999)u
Sixth Sense, The (1999)u

generate与evaluate所用的lora_weight是一样的，都是用自己前面那个训练出来的权重值，但他们的输出在两个py文件下不一样。

第二个样本：
-------label-------
Gods Must Be Crazy, The (1980)
--------res---------
['Lord of the Rings: The Fellowship of the Ring, The (2001)Notting Hill (1999)', 'Lord of the Rings: The Fellowship of the Ring, The (2001)+Notting Hill (1999)', 'Lord of the Rings: The Fellowship of the Ring, The (2001)Notting Hill (1999)', 'Lord of the Rings: The Fellowship of the Ring, The (2001)intl. (a.k.a. Lord of the Rings, The)', 'Lord of the Rings: The Fellowship of the Ring, The (2001)Here Comes the Boom (2012)', 'Lord of the Rings: The Fellowship of the Ring, The (2001)intl. (a.k.a. Lord of the Rings, The: The Fellowship of the Ring, The) (2001)', 'Lord of the Rings: The Fellowship of the Ring, The (2001)intl. (a.k.a. Lord of the Rings, The)', 'Lord of the Rings: The Fellowship of the Ring, The (2001)intl. (a.k.a. Lord of the Rings: The Fellowship of the Ring, The) (2001)', 'Lord of the Rings: The Fellowship of the Ring, The (2001)intl. (a.k.a. Lord of the Rings, The) (2001)', 'Lord of the Rings: The Fellowship of the Ring, The (2001)Notorious (1946)']
------inputs-------

generate.py文件产生的output：
Star Wars: Episode VI - Return of the Jedi (1983)Notorious (1946)
Star Wars: Episode VI - Return of the Jedi (1983)Notting Hill (1999)
Star Wars: Episode VI - Return of the Jedi (1983)er, Star Wars: Episode IV - A New Hope (1977)
Star Wars: Episode VI - Return of the Jedi (1983)er, Star Wars: Episode IV - A New Hope (1977)
![图片4](https://github.com/rutgerswiselab/GenRec/assets/149983492/6a914307-7b43-4862-ad48-2e0209dc1fa6)
![图片5](https://github.com/rutgerswiselab/GenRec/assets/149983492/70407367-96ea-4a8e-9c46-0d8a9f724691)

**Comment by jianchaoji** [comment]
如果你不加载LoRA，输出的结果是什么样的

**Comment by Quan-Yao** [comment]
> 如果你不加载LoRA，输出的结果是什么样的

不加载LoRA是什么意思，可以说下具体操作吗

**Comment by jianchaoji** [comment]
LoRA的原理是在原本的模型旁边加一个adapter，你的输出像是已经training成功了，但是evaluation里面adapter没加载上。所以你在模型加载的时候把model = PeftModel.from_pretrained 相关的注释掉，看看输出结果会不会有变化，如果没有变化，说明你可能package或者是什么地方没弄对，导致LoRA adapter没加载上

**Comment by Quan-Yao** [comment]
> LoRA的原理是在原本的模型旁边加一个adapter，你的输出像是已经training成功了，但是evaluation里面adapter没加载上。所以你在模型加载的时候把model = PeftModel.from_pretrained 相关的注释掉，看看输出结果会不会有变化，如果没有变化，说明你可能package或者是什么地方没弄对，导致LoRA adapter没加载上

我把model = PeftModel.from_pretrained相关的注释掉以后，它的输出结果是不同的，下面是不加adapter的输出：
![图片6](https://github.com/rutgerswiselab/GenRec/assets/149983492/2c963691-aecc-4512-80f9-44c4579c748d)

**Comment by Quan-Yao** [comment]
> > LoRA的原理是在原本的模型旁边加一个adapter，你的输出像是已经training成功了，但是evaluation里面adapter没加载上。所以你在模型加载的时候把model = PeftModel.from_pretrained 相关的注释掉，看看输出结果会不会有变化，如果没有变化，说明你可能package或者是什么地方没弄对，导致LoRA adapter没加载上
> 
> 我把model = PeftModel.from_pretrained相关的注释掉以后，它的输出结果是不同的，下面是不加adapter的输出： ![图片6](https://user-images.githubusercontent.com/149983492/282341343-2c963691-aecc-4512-80f9-44c4579c748d.png)

感觉还是adapter的问题，我加载你跑好的adapter，它的输出是正确的，我package的版本信息如下：[Here](https://github.com/Quan-Yao/GenRec_issues_2/blob/main/package)

**Comment by jianchaoji** [comment]
那可能还是训练过程中adapter没弄好，也可能是llama版本导致的问题
